sdk/core/core-tracing/test/instrumenter.spec.ts

```diff
@@ -72,6 +72,7 @@ describe("Instrumenter", () => {
       span.setStatus({ status: "success" });
       span.setAttribute("foo", "bar");
       span.recordException(new Error("test"));
+      span.addEvent!("I said span not Spren!", { startTime: new Date() });
```

Comment:
If we're making it optional, let's conditionally call it

```suggestion
      span.addEvent?.("I said span not Spren!", { startTime: new Date() });
```

Reply:
Yup fair point - I actually did this on purpose, because within the same package we expect the function to be defined (it may be undefined due to versioning)...

So, in this case, using the `!` operator allowed me to find a miss implementing this in the no-op span by yelling at me that it's undefined. Once I added `addEvent` to the no-op span the tests are happy again.

I can definitely see the tradeoff - do you think my reasoning makes sense or should I still change it to optional?

Reply:
maybe adding an assert for `span.addEvent` before calling it?

Reply:
Added assert.exists 👍